src/openloco/windows/vehicle.cpp

```diff
@@ -156,14 +156,7 @@ namespace openloco::ui::vehicle
             if (input::is_tool_active(WindowType::vehicle, w->number))
                 return;
 
-            registers regs;
-            regs.dx = 9; // tool widget
-            regs.al = 12;
-            regs.esi = (uintptr_t)w;
-
-            auto result = call(0x004CE367, regs);
-            bool cf = (result & (1 << 8)) != 0;
```

Comment:
What happened to the `(1 <<  8)` flag check?

Reply:
This seems to be just a boolean return. In the assembly code it was with a carry flag (stc instruction). cf seems to be also a carry flag, however I don't know why is it implemented this way.
I refactored the code from OpenRCT2 to suit this.

Reply:
Yes this is checking the carry flag. X86_FLAG_CARRY will be in bit zero of the ah register. Ax is copied to the `call` result which means it's actually at bit 8. CS often uses this for functions that return a bool. Your change looks correct.